python/tank/authentication/console_authentication.py

```diff
@@ -68,47 +79,149 @@ def authenticate(self, hostname, login, http_proxy):
-                    )
+                return auth_fn(hostname, login, http_proxy)
             except AuthenticationError as error:
                 # If any combination of credentials are invalid (user + invalid pass or
                 # user + valid pass + invalid 2da code) we'll end up here.
                 print("Login failed: %s" % error)
                 print()
 
+    def _authenticate_legacy(self, hostname, login, http_proxy):
+        # Get the credentials from the user
+        try:
+            hostname, login, password = self._get_user_credentials(
+                hostname, login, http_proxy
+            )
+        except EOFError:
+            # Insert a \n on the current line so the print is displayed on a new line.
+            print()
+            raise AuthenticationCancelled()
+
+        try:
+            # Try to generate a session token and return the user info.
+            return (
+                hostname,
+                login,
+                session_cache.generate_session_token(
+                    hostname, login, password, http_proxy
+                ),
+                None,
+            )
+        except MissingTwoFactorAuthenticationFault:
+            # session_token was None, we need 2fa.
+            code = self._get_2fa_code()
+            # Ask again for a token using 2fa this time. If this throws an AuthenticationError because
+            # the code is invalid or already used, it will be caught by the except clause beneath.
+            return (
+                hostname,
+                login,
+                session_cache.generate_session_token(
+                    hostname, login, password, http_proxy, auth_token=code
+                ),
+                None,
+            )
+
+    def _authenticate_unified_login_flow2(self, hostname, login, http_proxy):
+        print()
+        print(
+            "The authentication to {sg_url} requires to use your web browser.\n"
+            "\n"
+            'After pressing "continue", your web browser will open shortly '
+            "targeting your selected ShotGrid site.\n"
+            "\n"
+            "If you are not already authenticated to {sg_url} in the browser,"
+            "you will first need to authenticate.\n"
+            "\n"
+            "Then, you will be prompted to review the access request.\n"
+            'Select "Approve" and come back to this application.'
+            "\n".format(sg_url=hostname)
+        )
+
+        self._read_clean_input("Press enter when you are ready to continue")
+        print("\n")  # Always have 2 empty lines after a prompt
+        print(
+            "Your browser will open shortly.\n"
+            "Once you approved the access request, come back to this "
+            "application"
+        )
+        print()
+        session_info = ulf2_authentication.process(
+            hostname,
+            http_proxy=http_proxy,
+            product="toolkit",  # Same as "PRODUCT_IDENTIFIER" from LoginDialog
```

Comment:
Is it possible to use that already constant instead?

Reply:
This will be handle by SG-30915.